libraries/src/Document/HtmlDocument.php

```diff
@@ -559,8 +560,20 @@ public function getBuffer($type = null, $name = null, $attribs = array())
 		if ($this->_caching == true && $type === 'modules' && $name !== 'debug')
 		{
 			/** @var  \Joomla\CMS\Document\Renderer\Html\ModulesRenderer  $renderer */
-			$cache = CmsFactory::getCache('com_modules', '');
-			$hash = md5(serialize(array($name, $attribs, null, get_class($renderer))));
+			/** @var  \Joomla\CMS\Cache\Controller\OutputController  $cache */
+			$cache = CmsFactory::getContainer()->get(CacheControllerFactoryInterface::class)
+				->createCacheController('output', ['defaultgroup' => 'com_modules']);
+			$hash = md5(
+				serialize(
+					[
+						$name,
+						$attribs,
+						null,
+						get_class($renderer),
```

Comment:
Some functions in namespaced files should be root namespaced with `\` for "performance" as per https://github.com/php/php-src/blob/f2db305fa4e9bd7d04d567822687ec714aedcdb5/Zend/zend_compile.c#L3872

```suggestion
						\get_class($renderer),
```